src/types/error.ts

```diff
@@ -2,8 +2,30 @@
  * This file extend the base Error with a statusCode field.
  */
 
-interface IDetailedError {
-  readonly statusCode: number;
+import {
+  ResponseErrorInternal,
+  ResponseErrorNotFound
+} from "italia-ts-commons/lib/responses";
+
+interface IAPIError {
+  // tslint:disable-next-line:no-any
+  readonly toHTTPError: () => any;
 }
 
-export type APIError = IDetailedError & Error;
+export type APIError = IAPIError & Error;
+
+export function internalError(message: string): APIError {
```

Comment:
I'm not sure about the usefulness of this function, what do we lose by using directly `ResponseErrorInternal` instead of `internalError`?